src/libstd/io/mod.rs

```diff
@@ -219,6 +318,36 @@ pub trait Read {
     ///
     /// The returned adaptor also implements `Read` and will simply borrow this
     /// current reader.
+    ///
+    /// # Examples
+    ///
+    /// [`File`][file]s implement `Read`:
+    ///
+    /// [file]: ../std/fs/struct.File.html
+    ///
+    /// ```
+    /// use std::io;
+    /// use std::io::prelude::*;
+    /// use std::fs::File;
+    ///
+    /// # fn foo() -> io::Result<()> {
+    /// let mut f = try!(File::open("foo.txt"));
+    /// let mut buffer = Vec::new();
+    /// let mut other_buffer = Vec::new();
+    ///
+    /// {
+    ///     let reference = <File as Read>::by_ref(&mut f);
```

Comment:
Yeah this is... an unfortunate part about these two methods. I'd prefer to kinda just paper over this by actually breaking the convention a bit and sneakily only importing `std::io::Read` at the top instead of the whole prelude. That way only one method is in scope and you can call `.by_ref()` as a method. 

For example if we're resorting to this kind of ergonomics it would be preferrable to do:

```rust
{ 
    let reference = &mut f;
    try!(reference.take(...) ... );
}
```

(e.g. `&mut R` implements read and is exactly what `by_ref` does)